wfe/wfe.go

```diff
@@ -329,6 +331,54 @@ func (wfe *WebFrontEndImpl) handleKey(
 	}
 }
 
+func (wfe *WebFrontEndImpl) handleCertStatusBySerial(
+	ctx context.Context,
+	response http.ResponseWriter,
+	request *http.Request) {
+
+	serialStr := strings.TrimPrefix(request.URL.Path, certStatusBySerial)
+	serial := big.NewInt(0)
+	if _, ok := serial.SetString(serialStr, 16); !ok {
+		response.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	var status string
+	var cert *core.Certificate
+	var rcert *core.RevokedCertificate
+	if cert = wfe.db.GetCertificateBySerial(serial); cert != nil {
+		status = "Valid"
+	}
+	if rcert = wfe.db.GetRevokedCertificateBySerial(serial); rcert != nil {
+		status = "Revoked"
+		cert = rcert.Certificate
+	}
+
+	if status == "" {
+		response.WriteHeader(http.StatusNotFound)
+		return
+	}
+	result := make(map[string]interface{})
```

Comment:
In this scenario I'd generally create an anonymous type instead of a map of `interface{}`. It gives you slightly better type safety and I think is a little bit more idiomatic for Go.

Something like:
```
result := struct{
  Status string
  Serial string
  Certificate sttring
} {
  Status: status,
  Serial: serial.Text(16),
  Certificate: string(cert.PEM()),
}
....
```

Reply:
Done! Thanks for this as well.

Adding `Reason` and `RevokedAt` was more tricky; it turned out that `json:"omitempty"` doesn't output anything ever, and I needed to use `json:",omitempty"`. The comma turned out to be very important :)

Reply:
Doh! I'm 90% sure I've fallen into that trap before myself. I conveniently left out those fields in my suggested code :laughing: I'm glad you figured it out!